### ISSUE TITLE
tp_print is deprecated error

### ISSUE THREAD

**Issue by shyamsantoki** [pandas-dev/pandas#35618]
# tp_print is deprecated
I'm getting ```tp_print is deprecated``` whenever I try to install Pandas on Termux.
Here's few lines of error I'm getting
```
/data/data/com.termux/files/usr/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated he
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /data/data/com.termux/files/usr/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  pandas/io/sas/sas.c:20655:3: error: 'tp_print' is deprecated [-Werror,-Wdeprecated-declarations]
    0, /*tp_print*/
    ^
  /data/data/com.termux/files/usr/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated he
      Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
      ^
  /data/data/com.termux/files/usr/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
  #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                       ^
  5 errors generated.
  building 'pandas._libs.json' extension
  creating build/temp.linux-aarch64-3.8/pandas/_libs/src/ujson
  creating build/temp.linux-aarch64-3.8/pandas/_libs/src/ujson/python
  creating build/temp.linux-aarch64-3.8/pandas/_libs/src/ujson/lib
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-st
  aarch64-linux-android-clang -shared -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -Wl,--enab
  error: command 'aarch64-linux-android-clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pandas
Failed to build pandas
```

However rolling back cython didn't work for me, I was still getting error. So I tried to rollback Pandas to 1.0.5 and it worked. So I think there's some bug in Pandas package from your side which results above error.

```
pip list
Package            Version
------------------ ---------
appdirs            1.4.4
argon2-cffi        20.1.0
asv                0.4.2
attrs              19.3.0
backcall           0.2.0
black              19.10b0
bleach             3.1.5
certifi            2020.6.20
cffi               1.14.1
chardet            3.0.4
click              7.1.2
Cython             0.29.21
decorator          4.4.2
defusedxml         0.6.0
entrypoints        0.3
idna               2.10
ipykernel          5.3.4
ipython            7.17.0
ipython-genutils   0.2.0
ipywidgets         7.5.1
jedi               0.17.2
Jinja2             2.11.2
jsonschema         3.2.0
jupyter            1.0.0
jupyter-client     6.1.6
jupyter-console    6.1.0
jupyter-core       4.6.3
lxml               4.5.2
MarkupSafe         1.1.1
mistune            0.8.4
nbconvert          5.6.1
nbformat           5.0.7
notebook           6.1.1
numpy              1.19.1
packaging          20.4
pandas             1.0.5
pandocfilters      1.4.2
parso              0.7.1
pathspec           0.8.0
pexpect            4.8.0
pickleshare        0.7.5
pip                20.2.1
prometheus-client  0.8.0
prompt-toolkit     3.0.5
ptyprocess         0.6.0
pycparser          2.20
Pygments           2.6.1
pymongo            3.11.0
pyparsing          2.4.7
pyrsistent         0.16.0
python-dateutil    2.8.1
pytz               2020.1
pyzmq              19.0.2
qtconsole          4.7.5
QtPy               1.9.0
regex              2020.7.14
requests           2.24.0
Send2Trash         1.5.0
setuptools         49.2.1
six                1.15.0
terminado          0.8.3
testpath           0.4.4
toml               0.10.1
tornado            6.0.4
traitlets          4.3.3
typed-ast          1.4.1
urllib3            1.25.10
wcwidth            0.2.5
webencodings       0.5.1
wheel              0.34.2
widgetsnbextension 3.5.1
```

**Comment by TomAugspurger** [comment]
I think https://github.com/cython/cython/issues/3474 is the issue to follow here. Not sure, but I don't know that there's anything pandas can do @shyamsantoki.

**Comment by simonjayhawkins** [comment]
> I think [cython/cython#3474](https://github.com/cython/cython/issues/3474) is the issue to follow here. Not sure, but I don't know that there's anything pandas can do @shyamsantoki.

@TomAugspurger so can be closed as a duplicate of #33239

**Comment by TomAugspurger** [comment]
Yeah I think so.
